# HG changeset patch
# User Frédéric Wang <fred.wang@free.fr>
# Parent  d3228c82badd270863c56fc6b395e1ff452bb99a
Reset CSS spacing properties on the <math> element. b=1187673, r=jkitch

diff --git a/layout/mathml/mathml.css b/layout/mathml/mathml.css
--- a/layout/mathml/mathml.css
+++ b/layout/mathml/mathml.css
@@ -16,16 +16,19 @@
 math {
   writing-mode: horizontal-tb !important;
   direction: ltr;
   unicode-bidi: embed;
   display: inline;
   font-size: inherit;
   font-style: normal;
   font-family: serif;
+  line-height: normal;
+  word-spacing: normal;
+  letter-spacing: normal;
   text-rendering: optimizeLegibility;
   -moz-float-edge: margin-box;
   -moz-math-display: inline;
 }
 math[mode="display"], math[display="block"] {
   display: block;
   text-align: -moz-center;
   -moz-math-display: block;
diff --git a/layout/reftests/mathml/css-spacing-1-ref.html b/layout/reftests/mathml/css-spacing-1-ref.html
new file mode 100644
--- /dev/null
+++ b/layout/reftests/mathml/css-spacing-1-ref.html
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>CSS spacing properties</title>
+    <style type="text/css">
+      math {
+        line-height: normal;
+        word-spacing: normal;
+        letter-spacing: normal;
+      }
+    </style>
+  </head>
+
+  <body>
+    <p style="line-height: 30pt;
+              word-spacing: 2em;
+              letter-spacing: 2em;">
+      <math xmlns="http://www.w3.org/1998/Math/MathML" display="block">
+        <mrow>
+          <mo>(</mo>
+          <mtable rowspacing="0pt" columnspacing="0em">
+            <mtr>
+              <mtd><mtext>row 1 column 1</mtext></mtd>
+              <mtd><mtext>row 1 column 2</mtext></mtd>
+            </mtr>
+            <mtr>
+              <mtd><mtext>row 2 column 1</mtext></mtd>
+              <mtd><mtext>row 2 column 2</mtext></mtd>
+            </mtr>
+          </mtable>
+          <mo>)</mo>
+        </mrow>
+      </math>
+    </p>
+  </body>
+</html>
diff --git a/layout/reftests/mathml/css-spacing-1.html b/layout/reftests/mathml/css-spacing-1.html
new file mode 100644
--- /dev/null
+++ b/layout/reftests/mathml/css-spacing-1.html
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>CSS spacing properties</title>
+  </head>
+
+  <body>
+    <p style="line-height: 30pt;
+              word-spacing: 2em;
+              letter-spacing: 2em;">
+      <math xmlns="http://www.w3.org/1998/Math/MathML" display="block">
+        <mrow>
+          <mo>(</mo>
+          <mtable rowspacing="0pt" columnspacing="0em">
+            <mtr>
+              <mtd><mtext>row 1 column 1</mtext></mtd>
+              <mtd><mtext>row 1 column 2</mtext></mtd>
+            </mtr>
+            <mtr>
+              <mtd><mtext>row 2 column 1</mtext></mtd>
+              <mtd><mtext>row 2 column 2</mtext></mtd>
+            </mtr>
+          </mtable>
+          <mo>)</mo>
+        </mrow>
+      </math>
+    </p>
+  </body>
+</html>
diff --git a/layout/reftests/mathml/reftest.list b/layout/reftests/mathml/reftest.list
--- a/layout/reftests/mathml/reftest.list
+++ b/layout/reftests/mathml/reftest.list
@@ -5,16 +5,17 @@
 == dir-5.html dir-5-ref.html
 == dir-6.html dir-6-ref.html
 == dir-6a.html dir-6a-ref.html
 == dir-7.html dir-7-ref.html
 fails == dir-8.html dir-8-ref.html
 fails == dir-9.html dir-9-ref.html # Bug 787215
 == dir-10.html dir-10-ref.html
 random-if((B2G&&browserIsRemote)||Mulet) == dir-11.html dir-11-ref.html # Initial mulet triage: parity with B2G/B2G Desktop
+== css-spacing-1.html css-spacing-1-ref.html
 == displaystyle-1.html displaystyle-1-ref.html
 == displaystyle-2.html displaystyle-2-ref.html
 == displaystyle-3.html displaystyle-3-ref.html
 random-if((B2G&&browserIsRemote)||Mulet) == displaystyle-4.html displaystyle-4-ref.html # Initial mulet triage: parity with B2G/B2G Desktop
 skip-if((B2G&&browserIsRemote)||Mulet) random-if(smallScreen&&Android) fuzzy(255,200) == mirror-op-1.html mirror-op-1-ref.html # Initial mulet triage: parity with B2G/B2G Desktop
 != mirror-op-2.html mirror-op-2-ref.html
 != mirror-op-3.html mirror-op-3-ref.html
 != mirror-op-4.html mirror-op-4-ref.html
